### PR TITLE
Adding mkdocstrings

### PR DESCRIPTION
**Pull request by oboulant**

### CODE REVIEW

**Comment by deepcharles** [comment]
Ok, it looks nice. I will organize the content in the documentation.

**Comment by oboulant** [comment]
@deepcharles FYI I am doing `base.py` and `costs/costautoregressive.py`at the moment

**Comment by oboulant** [comment]
Still have to deal with citation and reference to some .bib files

File: docs/user-guide/cost-functions/costautoregressive.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+Consider the following piecewise autoregressive model
+
+$$
+    y_t = z_t' \delta_j + \varepsilon_t, \quad \forall t=t_j,\dots,t_{j+1}-1
+$$
+
+where $j>1$ is the segment number, $z_t=[y_{t-1}, y_{t-2},\dots,y_{t-p}]$ is the lag vector,and $p>0$ is the order of the process.
+
+The least-squares estimates of the break dates is obtained by minimizing the sum of squared
+residuals :cite:`ar-Bai2000`. <!--- How to handle citation ?  -->
```
Comment (oboulant) [inline docs/user-guide/cost-functions/costautoregressive.md:15]: @deepcharles still have to find out how to handle this. Maybe it is already done in what you did in the other user guide entries. Will have a look !

Comment (deepcharles) [reply docs/user-guide/cost-functions/costautoregressive.md:15]: I made a simple Markdown link, formatted like a citation. See https://github.com/deepcharles/ruptures/blob/mkdocs_code_reference/docs/user-guide/cost-functions/costnormal.md for an example.

**Pull request merged**